src/main/resources/META-INF/plugin.xml

```diff
@@ -368,6 +368,12 @@ L
                 order="first"
                 implementation="com.redhat.devtools.lsp4ij.features.formatting.LSPServerSideOnTypeFormattingEnterHandler"/>
 
+        <!-- Client-side on-type formatting -->
+        <typedHandler
+                id="LSPClientSideOnTypeFormattingTypedHandler"
+                order="first, before LSPServerSideOnTypeFormattingTypedHandler"
```

Comment:
Note that I've ordered this before server-side on-type formatting. That way if the server supports on-type formatting and the user prefers client-side on-type formatting for one of the supported characters, they just need to enable that feature in client config and client-side on-type formatting will supersede server-side on-type formatting because, when a character is handled, `Result.STOP` is returned and no further `typedHandler` implementations are called.